examples/uxmal/uxmal.cpp

```diff
@@ -0,0 +1,212 @@
+                  _xfm(f.getLeft() * f.getFar() / f.getNear(),
+                       f.getBottom() * f.getFar() / f.getNear(), -f.getFar(),
+                       f.getView()),
+                  _xfm(f.getRight() * f.getFar() / f.getNear(),
+                       f.getBottom() * f.getFar() / f.getNear(), -f.getFar(),
+                       f.getView()),
+                  _xfm(f.getRight() * f.getFar() / f.getNear(),
+                       f.getTop() * f.getFar() / f.getNear(), -f.getFar(),
+                       f.getView()),
+                  _xfm(f.getLeft() * f.getFar() / f.getNear(),
+                       f.getTop() * f.getFar() / f.getNear(), -f.getFar(),
+                       f.getView())}
+    {
+    }
+
+    explicit Box(const eq::uint128_t& i, const AABB& b)
+        : id(i)
+        , corners{{b.getMin()[0], b.getMin()[1], b.getMin()[2]},
+                  {b.getMin()[0], b.getMax()[1], b.getMin()[2]},
+                  {b.getMax()[0], b.getMax()[1], b.getMin()[2]},
+                  {b.getMax()[0], b.getMin()[1], b.getMin()[2]},
+                  {b.getMin()[0], b.getMin()[1], b.getMax()[2]},
+                  {b.getMin()[0], b.getMax()[1], b.getMax()[2]},
+                  {b.getMax()[0], b.getMax()[1], b.getMax()[2]},
+                  {b.getMax()[0], b.getMin()[1], b.getMax()[2]}}
+    {
+    }
+
+    const eq::uint128_t id;
+    const eq::Vector3f corners[8];
+};
+
+class Renderer : public seq::Renderer
+{
+    template <class O>
+    void _handle(const void* data, const size_t size)
+    {
+        std::unique_ptr<const O> object = O::create(data, size);
+
+        if (object->getFrame() > _dataFrame)
+        {
+            _dataFrame = object->getFrame();
+            _boxes.clear();
+            _ids.clear();
+        }
+
+        if (_dataFrame > object->getFrame() + 10) // restarted app, reset
+            _dataFrame = object->getFrame();
+
+        if (_dataFrame != object->getFrame())
+            return;
+
+        auto i = _ids.find(object->getId());
+        if (i == _ids.end())
+            i = _ids.insert(object->getId()).first;
+        _boxes.emplace_back(Box{object->getId(), *object});
+    }
+
+public:
+    Renderer(seq::Application& application)
+        : seq::Renderer(application)
+    {
+        _subscriber.subscribe(Frustum::ZEROBUF_TYPE_IDENTIFIER(),
+                              [this](const void* data, const size_t size) {
+                                  _handle<Frustum>(data, size);
+                              });
+        _subscriber.subscribe(AABB::ZEROBUF_TYPE_IDENTIFIER(),
+                              [this](const void* data, const size_t size) {
+                                  _handle<AABB>(data, size);
+                              });
+    }
+    virtual ~Renderer() {}
+protected:
+    bool init(co::Object*) final
+    {
+        glEnable(GL_LINE_SMOOTH);
+        glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
```

Comment:
funny, I looked that up too recently. Turns out it does not work (well) on some machines like the cluster iirc. anti-aliased lines are still impossible in 2017 :p